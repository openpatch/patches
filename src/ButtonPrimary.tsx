/** @jsx jsx */
import { ButtonHTMLAttributes, ReactNode } from "react";
import { ButtonBase, ButtonBaseProps } from "./ButtonBase";
import styled from "./themes/styled";
import { variant } from "./system";
import { SystemColor, SystemColorNames } from "./system/types";

export type ButtonPrimaryProps = {
  children: string;
  tone?: SystemColorNames;
} & ButtonHTMLAttributes<HTMLButtonElement> &
  Pick<
    ButtonBaseProps,
    | "iconLeft"
    | "iconRight"
    | "size"
    | "disabled"
    | "type"
    | "fullWidth"
    | "loading"
  >;

const StyledButtonPrimary: React.FunctionComponent<ButtonPrimaryProps> = styled(
  ButtonBase
)<ButtonPrimaryProps>(
  {},
  variant<
    {
      color: SystemColor;
      backgroundColor: SystemColor;
    },
    SystemColorNames,
    string
  >({
    scale: "buttonsPrimary",
    prop: "tone",
    variants: {
      primary: {
        backgroundColor: "primary.800",
        color: "primary.50",
      },
      accent: {
        backgroundColor: "accent.800",
        color: "accent.50",
      },
      neutral: {
        backgroundColor: "neutral.800",
        color: "neutral.50",
      },
      success: {
        backgroundColor: "success.800",
        color: "success.50",
      },
      warning: {
        backgroundColor: "warning.800",
        color: "warning.50",
      },
      error: {
        backgroundColor: "error.800",
        color: "error.50",
      },
      info: {
        backgroundColor: "info.800",
        color: "info.50",
      },
    },
  })
);

export const ButtonPrimary = StyledButtonPrimary;
ButtonPrimary.defaultProps = {
  tone: "primary",
};
