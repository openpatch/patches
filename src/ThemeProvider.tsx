import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
