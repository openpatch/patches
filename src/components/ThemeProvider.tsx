import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
