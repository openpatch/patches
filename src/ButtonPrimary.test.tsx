import React from "react";
import { render } from "@testing-library/react";
import { ButtonPrimary, ButtonPrimaryProps } from "./ButtonPrimary";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: ButtonPrimaryProps) =>
  render(
    <ThemeProvider>
      <ButtonPrimary {...props}>Button</ButtonPrimary>
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
