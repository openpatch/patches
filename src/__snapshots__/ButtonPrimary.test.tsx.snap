// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-1 {
  position: relative;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 8px;
  font-weight: 600;
  -webkit-transition: all 0.1s ease-out;
  transition: all 0.1s ease-out;
  font-size: 14px;
  padding-left: 10px;
  padding-right: 10px;
  padding-top: 8px;
  padding-bottom: 8px;
  background-color: #0C6B58;
  color: #EFFCF6;
}

.emotion-1:disabled {
  cursor: not-allowed;
  opacity: 0.4;
}

.emotion-1:hover:enabled {
  opacity: 0.8;
}

.emotion-1:active:enabled {
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(66,153,225,0.5);
}

.emotion-1 span.icon-left,
.emotion-1 span.icon-right {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-1 span.icon-left,
.emotion-1 span.icon-right {
  width: 14px;
}

.emotion-1 span.icon-left {
  margin-right: 8px;
}

.emotion-1 span.icon-right {
  margin-left: 8px;
}

.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  cursor: pointer;
  border-radius: 8px;
  font-weight: 600;
  -webkit-transition: all 0.1s ease-out;
  transition: all 0.1s ease-out;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0 0 0 3px rgba(66,153,225,0.5);
}

.emotion-0 span.icon-left,
.emotion-0 span.icon-right {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<div>
  <button
    class="emotion-1"
  >
    <div
      class="emotion-0"
    >
      Button
    </div>
  </button>
</div>
`;
